packages/edit-site/src/components/page-templates/template-actions.js

```diff
@@ -0,0 +1,228 @@
+
+export function useResetTemplateAction() {
+	const { revertTemplate } = useDispatch( editSiteStore );
+	const { saveEditedEntityRecord } = useDispatch( coreStore );
+	const { createSuccessNotice, createErrorNotice } =
+		useDispatch( noticesStore );
+	return useMemo(
+		() => ( {
+			id: 'reset-template',
+			label: __( 'Reset template' ),
+			isPrimary: true,
+			icon: backup,
+			isEligible: isTemplateRevertable,
+			async perform( template ) {
+				try {
+					await revertTemplate( template, { allowUndo: false } );
+					await saveEditedEntityRecord(
+						'postType',
+						template.type,
+						template.id
+					);
+
+					createSuccessNotice(
+						sprintf(
+							/* translators: The template/part's name. */
+							__( '"%s" reverted.' ),
+							decodeEntities( template.title.rendered )
+						),
+						{
+							type: 'snackbar',
+							id: 'edit-site-template-reverted',
+						}
+					);
+				} catch ( error ) {
+					const fallbackErrorMessage =
+						template.type === TEMPLATE_POST_TYPE
+							? __(
+									'An error occurred while reverting the template.'
+							  )
+							: __(
+									'An error occurred while reverting the template part.'
+							  );
+					const errorMessage =
+						error.message && error.code !== 'unknown_error'
+							? error.message
+							: fallbackErrorMessage;
+
+					createErrorNotice( errorMessage, { type: 'snackbar' } );
+				}
+			},
+		} ),
+		[
+			createErrorNotice,
+			createSuccessNotice,
+			revertTemplate,
+			saveEditedEntityRecord,
+		]
+	);
+}
+
+export const deleteTemplateAction = {
+	id: 'delete-template',
+	label: __( 'Delete template' ),
+	isPrimary: true,
+	icon: trash,
+	isEligible: isTemplateRemovable,
+	modalProps: {
+		ModalContent: ( { item: template, setIsModalOpen } ) => {
+			const { removeTemplate } = useDispatch( editSiteStore );
+			return (
+				<VStack spacing="5">
+					<Text>
+						{ sprintf(
+							// translators: %s: The template or template part's title.
+							__( 'Are you sure you want to delete "%s"?' ),
+							decodeEntities( template.title.rendered )
+						) }
+					</Text>
+					<HStack justify="right">
+						<Button
+							variant="tertiary"
+							onClick={ () => {
+								setIsModalOpen( false );
+							} }
+						>
+							{ __( 'Cancel' ) }
+						</Button>
+						<Button
+							variant="primary"
+							onClick={ () =>
+								removeTemplate( template, {
+									allowUndo: false,
+								} )
+							}
+						>
+							{ __( 'Delete' ) }
+						</Button>
+					</HStack>
+				</VStack>
+			);
+		},
+	},
+};
+
+export const renameTemplateAction = {
+	id: 'rename-template',
+	label: __( 'Rename' ),
+	isEligible: ( template ) =>
+		isTemplateRemovable( template ) && template.is_custom,
+	modalProps: {
+		// TODO: should we restrict this API to only allow some of Modal props?
+		title: __( 'Rename' ),
+		overlayClassName: 'edit-site-list__rename-modal',
+		ModalContent: ( { item: template, setIsModalOpen } ) => {
```

Comment:
Maybe that's the only config we need. like instead of `perform` you define `renderModal` (this function) and that's it?

Reply:
What do you mean here?